src/h3lib/lib/polygon.c

```diff
@@ -93,3 +95,192 @@ bool pointInsidePolygon(const GeoPolygon* geoPolygon, const BBox* bboxes,
 #undef IS_EMPTY
 #undef INIT_ITERATION
 #undef ITERATE
+
+/**
+ * Count the number of polygons containing a given loop.
+ * TODO: It would be really nice to memoize this.
+ * @param  loop         Loop to count containers for
+ * @param  polygons     Polygons to test
+ * @param  polygonCount Number of polygons in the test array
+ * @return              Number of polygons containing the loop
+ */
+static int countContainers(const LinkedGeoLoop* loop,
+                           LinkedGeoPolygon** polygons, const BBox** bboxes,
+                           const int polygonCount) {
+    int containerCount = 0;
+    for (int i = 0; i < polygonCount; i++) {
+        if (loop != polygons[i]->first &&
+            pointInsideLinkedGeoLoop(polygons[i]->first, bboxes[i],
+                                     &loop->first->vertex)) {
+            containerCount++;
+        }
+    }
+    return containerCount;
+}
+
+/**
+ * Given a list of nested containers, find the one most deeply nested.
+ * @param  polygons     Polygon containers to check
+ * @param  bboxes       Bounding boxes for polygons, used in point-in-poly check
+ * @param  polygonCount Number of polygons in the list
+ * @return              Deepest container, or null if list is empty
+ */
+static LinkedGeoPolygon* findDeepestContainer(LinkedGeoPolygon** polygons,
```

Comment:
OK I understand the motivation to use non-`const` for both return type and polygons, but it makes it seem like you modify them in this function, which as far as I can tell, isn't the case. Maybe return `const LinkedGeoPolygon*` and just cast away the `const` in the calling code. This is pretty standard C style when you have an unnecessary `const`. C++ is a little better about it because you can have `const` and non-`const` overloads of the same function, but since C does not, it is accepted to cast away the `const`.

My obsession with `const` is from what I have heard that using `const` wherever possible is better practice and allows for better compiler optimizations.

Reply:
What's the syntax for casting it away? The issue is that I pass in polygons, get one of them back, and then modify it. If I say it's `const` in the param list, then I have to return it as `const`, and modification is a compile error. Happy to have it treated as const in the inner functions if there's a way to do so.

Reply:
Nvm, got it and added the `const` qualifiers.